src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,270 @@
+        /// The DiagnosticsContext class constructor.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="handle">Diagnostics context received in the ReportArrived event.</param>
+        public DiagnosticsContext(IntPtr handle)
+        {
+            _handle = handle;
+
+            var ret = NativeApi.GetType(handle, out NativeApi.CtxType contextType);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            Type = (CtxType)contextType;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the DiagnosticsContext class.
+        /// </summary>
+        ~DiagnosticsContext()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets report of specific type from the context.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="report">Report type to retrieve from the context.</param>
+        /// <feature>
+        /// http://tizen.org/privilege/internal/default/platform
+        /// </feature>
+        public DiagnosticsData GetReport(ReportType report)
+        {
+            var ret = NativeApi.GetReport(_handle, (NativeApi.ReportType)report, out IntPtr data);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new DiagnosticsData(data);
+        }
+
+        /// <summary>
+        /// Stores context type.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public CtxType Type;
+
+        /// <summary>
+        /// Dispose API for closing the internal resources.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Dispose API for closing the internal resources.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (_handle != IntPtr.Zero)
+                {
+                    NativeApi.Destroy(_handle);
+                    _handle = IntPtr.Zero;
+                }
+                _disposed = true;
+            }
+        }
+    }
+
+    /// <summary>
+    /// The DiagnosticsData class is used for reading data (either from report or client).
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class DiagnosticsData : IDisposable
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// The DiagnosticsData class constructor.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="handle">Diagnostics data handle.</param>
+        public DiagnosticsData(IntPtr handle)
```

Comment:
Do not open API using IntPtr. If you need IntPtr in the public API, consider using SafeHandle instead.
```suggestion
        internal DiagnosticsData(IntPtr handle)
```